lib/logstash/outputs/elasticsearch/common.rb

```diff
@@ -138,6 +138,7 @@ def event_action_params(event)
       if @action == 'update'
         params[:_upsert] = LogStash::Json.load(event.sprintf(@upsert)) if @upsert != ""
         params[:_script] = event.sprintf(@script) if @script != ""
+        params[:_retry_on_conflict] = @retry_on_conflict
```

Comment:
This is different because we use the bulk API. These options are passed directly per https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-bulk.html#bulk-update

Note the underscore there.